File: 1-easy/problems/0094-binary-tree-inorder-traversal/solution.py
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        def dfs(node):
            if not node: return
            dfs(node.left); res.append(node.val); dfs(node.right)
        dfs(root)
        return res

File: 1-easy/problems/0094-binary-tree-inorder-traversal/solution.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> res;
        function<void(TreeNode*)> dfs = [&](TreeNode* node) {
            if (!node) return;
            dfs(node->left); res.push_back(node->val); dfs(node->right);
        };
        dfs(root);
        return res;
    }
};
